Makefile

```diff
@@ -20,4 +22,7 @@ test_unittest:
 
 test: test_unittest test_integration
 
+coverage_report: test
+	coverage report -m
```

Comment:
Whether subprocesses are tracked? Are you plan to measuse coverage using only unit tests?

Reply:
No, I have changed patch to gather code coverage data with all tests.
Also added a patch to enable `coverage.py` for subprocesses:
```
--- a/lib/tarantool_server.py
+++ b/lib/tarantool_server.py
@@ -864,6 +864,12 @@ class TarantoolServer(Server):
 
         # redirect stdout from tarantoolctl and tarantool
         os.putenv("TEST_WORKDIR", self.vardir)
+        if os.environ.get('PYTHON') == 'coverage run':
+            import coverage
+            coveragerc_path = os.path.join(os.path.abspath(self.sourcedir), ".coveragerc")
+            assert(os.path.exists(coveragerc_path))
+            os.environ['COVERAGE_PROCESS_START'] = os.path.join(coveragerc_path)
+            coverage.process_startup()
         self.process = subprocess.Popen(args,
                                         cwd=self.vardir,
                                         stdout=self.log_des,
```

Reply:
I was in doubts about test-run workers (which are separate processes). Maybe it works by default, I don't know. Just asked.

Here we start a tarantool process, so I guess we should not track it in the coverage tool.

Reply:
Actually I don't know, but [this](https://coverage.readthedocs.io/en/latest/subprocess.html#subprocess) note looks relevant:

> If you have subprocesses because you are using `multiprocessing`, the `--concurrency=multiprocessing` command-line option should take care of everything for you. See [Execution: coverage run](https://coverage.readthedocs.io/en/latest/cmd.html#cmd-run) for details.

Reply:
The relevant configuration option is [run.concurrency](https://coverage.readthedocs.io/en/latest/config.html#config-run-concurrency). As I see, it allows to pass several values at once, so I would try to set both `multiprocessing` and `gevent`. At least it worth to try and look how results will be changed. (I guess that `gevent` assumes `greenlet`. According to the [source code](https://github.com/nedbat/coveragepy/blob/01cbb8751f98e5a7de79699444cbc03647691616/coverage/collector.py#L94-L97), they cannot be set both. It seems, we can combine `multiprocessing` with something else.)

Reply:
> The relevant configuration option is run.concurrency. As I see, it allows to pass several values at once, so I would try to set both multiprocessing and gevent. 

`multiprocessing` conflicts with `gevent`, so I kept `gevent` only. BTW no changes in overall code coverage number.

```
--- a/.coveragerc
+++ b/.coveragerc
@@ -1,5 +1,7 @@
 [run]
 branch = True
+concurrency =
+    gevent
 
 [report]
 precision = 1
```

Reply:
> > Here we start a tarantool process, so I guess we should not track it in the coverage tool.
> 
> python tests uses python helpers like `admin` etc so why we shouldn't want to track it?

We execute Python code in tests within a worker process using the [exec()](https://docs.python.org/3/library/functions.html#exec) function in `PythonTest`. Not here. Here we start tarantool processes.

Reply:
> `multiprocessing` conflicts with `gevent`, so I kept `gevent` only. BTW no changes in overall code coverage number.

Now I see that both are there. That was a mistake?

Reply:
Anyway, it seems, that the answer to my initial question is 'no': subprocesses are NOT tracked. I found that it is due to changing current directory to `test/`. So, the following patch does the work:

```diff
diff --git a/Makefile b/Makefile
index 4c5f72e..2b0361b 100644
--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ test: test_unittest test_integration
 
 coverage:
        PYTHON="coverage run" make -f $(MAKEFILE_PATH) test
-       coverage combine
+       coverage combine . test
        coverage report
 
 clean:
```

Reply:
added `coverage combine $(PROJECT_DIR) $(PROJECT_DIR)/test`

Reply:
>> multiprocessing conflicts with gevent, so I kept gevent only. BTW no changes in overall code coverage number.

> Now I see that both are there. That was a mistake?

Possibly I added `gevent` back when it started to work.

Reply:
Nice, thanks!